src/Transport/Http.php

```diff
@@ -87,7 +94,16 @@ public function exec(Request $request, array $params): Response
         /* @see Connection::setConnectTimeout() */
         $connectTimeout = $connection->getConnectTimeout();
         if ($connectTimeout > 0) {
-            \curl_setopt($conn, \CURLOPT_CONNECTTIMEOUT, $connectTimeout);
+            // If cURL >= 7.16.2, timeout can be expressed in ms
+            if (\defined('CURLOPT_CONNECTTIMEOUT_MS')) {
+                // since we're accepting float, and the option uses an int, let's round and use a
+                // minimum value of "1" for the option. "0" means "no timeout"
+                \curl_setopt($conn, \CURLOPT_CONNECTTIMEOUT_MS, \max(1, \round($connectTimeout * 1000)));
```

Comment:
Do we need to use `max` here, as there is already a check for only allowing value greater than `0`.

Reply:
yes, round(0.1) produces `0` whereas `0` as a different meaning

Reply:
Ok, so that's for avoiding rounding issue which could happen when using a value like `0.0001`.
Can't we make the code minimalist (for example as guzzle or symfony http client do) by just multiplying by 1000.
```php
\curl_setopt($conn, \CURLOPT_CONNECTTIMEOUT_MS, $connectTimeout * 1000);
```

Reply:
This constant is an int, not a float, see `https://curl.se/libcurl/c/CURLOPT_CONNECTTIMEOUT_MS.html`

Reply:
As a reminder, Symfony HTTP client does not provide interaction with `CURLOPT_CONNECTTIMEOUT_MS`, only for `CURLOPT_TIMEOUT`

As mentioned in the comment of this PR, passing `0` to `CURLOPT_CONNECTTIMEOUT_MS` is a complete different meaning than passing `0` to `CURLOPT_TIMEOUT`. In the latest case, it sets the value to `0`, in the first case it reset to the default connect timeout that equals to `300` in std cURL versions

Reply:
Yes but this would be automatically casted to long by using `zval_get_long`, which does under the hood an `zend_dval_to_lval` call when it's a double, so this doesn't require to pass an int exclusively.
See: 
- https://github.com/php/php-src/blob/master/ext/curl/interface.c#L2366

Just to confirm it works:
```php
<?php

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, "example.com");
curl_setopt($ch, CURLOPT_CONNECTTIMEOUT_MS, 0.1);
curl_setopt($ch, CURLOPT_TIMEOUT_MS, 0.1);
curl_exec($ch);
```